src/Config/Provider/AutodnsProvider.php

```diff
@@ -0,0 +1,234 @@
+                        'origin' => $ldhName,
+                        'main' => [
+                            'address' => $authData['dns_ip']
+                        ],
+                        'soa' => [
+                            'refresh' => 3600,
+                            'retry' => 7200,
+                            'expire' => 604800,
+                            'ttl' => 600
+                        ],
+                        'action' => 'COMPLETE',
+                        'wwwInclude' => true,
+                        'nameServers' => [
+                            [
+                                'name' => 'a.ns14.net'
+                            ],
+                            [
+                                'name' => 'b.ns14.net'
+                            ],
+                            [
+                                'name' => 'c.ns14.net'
+                            ],
+                            [
+                                'name' => 'd.ns14.net'
+                            ],
+                        ]
+
+                    ])
+                    ->toArray()
+            )->toArray();
+        }
+    }
+    
+    /**
+     * @throws TransportExceptionInterface
+     */
+    public static function verifyAuthData(array $authData, HttpClientInterface $client): array
+    {
+
+        $username = $authData['username'];
+        $password = $authData['password'];
+
+        $acceptConditions = $authData['acceptConditions'];
+        $ownerLegalAge = $authData['ownerLegalAge'];
+        $waiveRetractationPeriod = $authData['waiveRetractationPeriod'];
+
+        if (empty($authData['context'])) {
+            $authData['context'] = 4;
+        }
+
+        if (
+            empty($username) || empty($password)
+        ) {
+            throw new BadRequestHttpException('Bad authData schema');
+        }
+
+        if (
+            true !== $acceptConditions
+            || empty($authData['ownerConfirm'])
+            || true !== $ownerLegalAge
+            || true !== $waiveRetractationPeriod
+        ) {
+            throw new HttpException(451, 'The user has not given explicit consent');
+        }
+
+        try {
+            $response = $client->request(
+                'GET',
+                '/v1/hello',
+                (new HttpOptions())
+                    ->setAuthBasic($authData['username'], $authData['password'])
+                    ->setHeader('Accept', 'application/json')
+                    ->setHeader('X-Domainrobot-Context', $authData['context'])
+                    ->setBaseUri(self::BASE_URL)
+                    ->toArray()
+            );
+        } catch (\Exception $exp) {
+            throw new BadRequestHttpException('Invalid Login');
+        }
+
+        if (Response::HTTP_OK !== $response->getStatusCode()) {
+            throw new BadRequestHttpException('The status of these credentials is not valid');
+        }
+
+        return $authData;
```

Comment:
The `authData` object should be recreated here to avoid retaining unnecessary properties that the user might add for fun, which could mistakenly end up being stored in the database. Note that the structure of the authData object is specific to each Provider.

I also suggest storing the proofs of consent (`acceptConditions`, `ownerLegalAge`, `waiveRetractationPeriod`, `ownerConfirm`) in the database to ensure that there is a record of the consent.

```suggestion
        return [
            'username' => $username,
            'password' => $password,
            'ownerConfirm' => $authData['ownerConfirm'],
            'context' => $authData['context'],
            'contactid' => $authData['contactid'],
            'acceptConditions' => $acceptConditions,
            'ownerLegalAge' => $ownerLegalAge,
            'waiveRetractationPeriod' => $waiveRetractationPeriod,
        ];
```